processing/src/main/java/io/druid/query/groupby/orderby/DefaultLimitSpec.java

```diff
@@ -115,24 +115,32 @@ public int compare(Row left, Row right)
       }
     };
 
-    Map<String, Ordering<Row>> possibleOrderings = Maps.newTreeMap(String.CASE_INSENSITIVE_ORDER);
+    Map<String, DimensionSpec> dimensionsMap = Maps.newHashMap();
     for (DimensionSpec spec : dimensions) {
-      final String dimension = spec.getOutputName();
-      possibleOrderings.put(dimension, dimensionOrdering(dimension));
+      dimensionsMap.put(spec.getOutputName(), spec);
     }
 
+    Map<String, AggregatorFactory> aggregatorsMap = Maps.newHashMap();
     for (final AggregatorFactory agg : aggs) {
-      final String column = agg.getName();
-      possibleOrderings.put(column, metricOrdering(column, agg.getComparator()));
+      aggregatorsMap.put(agg.getName(), agg);
     }
 
+    Map<String, PostAggregator> postAggregatorsMap = Maps.newHashMap();
     for (PostAggregator postAgg : postAggs) {
-      final String column = postAgg.getName();
-      possibleOrderings.put(column, metricOrdering(column, postAgg.getComparator()));
+      postAggregatorsMap.put(postAgg.getName(), postAgg);
     }
 
     for (OrderByColumnSpec columnSpec : columns) {
-      Ordering<Row> nextOrdering = possibleOrderings.get(columnSpec.getDimension());
+      String columnName = columnSpec.getDimension();
+      Ordering<Row> nextOrdering = null;
+
+      if (dimensionsMap.containsKey(columnName)) {
+        nextOrdering = dimensionOrdering(columnName);
+      } else if (aggregatorsMap.containsKey(columnName)) {
+        nextOrdering = metricOrdering(columnName, aggregatorsMap.get(columnName).getComparator());
+      } else if (postAggregatorsMap.containsKey(columnName)) {
```

Comment:
This if-else chain effectively flips the order of precedence.

Prior behavior was to (in the event of equal named things) to favor post-agg over agg over dimension. This flips that behavior.

In the case where an aggregator and a post aggregator are both named `uniques` (for example, a hyper uniques aggregator and a hyper unique finalizer postAgg) the sorting will be based on the aggregator instead of the post-agg.

Is such a case covered by unit tests?